# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


In [2]:
import pandas as pd

### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [3]:
# Your code here
df = pd.read_csv('data\car-sales.csv')
df

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [8]:
# Your code here
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Manufacturer  10 non-null     object 
 1   Colour        10 non-null     object 
 2   Odometer      10 non-null     int64  
 3   Doors         10 non-null     int64  
 4   Price         10 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 528.0+ bytes


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [11]:
# Your code here
df.shape[0]

10

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [13]:
# Your code here
df['Manufacturer'].value_counts()

Toyota    4
Honda     3
Nissan    2
BMW       1
Name: Manufacturer, dtype: int64

### 5. Đếm số xe theo màu


In [14]:
# Your code here
df['Colour'].value_counts()

White    4
Blue     3
Black    1
Green    1
Red      1
Name: Colour, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [16]:
# Your code here
df['Price'].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [19]:
# Your code here
#ascending sắp xếp tăng
df.sort_values(by='Price', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [20]:
# Your code here
df.sort_values(by = 'Odometer', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [18]:
# Your code here
df.loc[(df.Manufacturer == 'Honda'), ['Price']].count()/10
df.loc[(df.Manufacturer == 'Honda'), ['Price']].mean()
df.loc[(df.Manufacturer == 'Honda'), ['Colour']].drop_duplicates()


Colour
1    Red
6   Blue

# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [25]:
# Your code here
# Giá trung bình các xe (7645)
print('Giá trung bình các xe: ', df['Price'].mean())
# - Giá trị trung vị các xe (6625)
print('Giá trị trung vị các xe: ', df['Price'].median())
# Miền giá trị của xe (18500)
print('Miền giá trị của mảng là: ', df['Price'].max() - df['Price'].min())
# - Miền phân vị của xe (2750)
print('Miền phân vị của xe: ', df['Price'].quantile([0.5]))
# - Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])
print('Điểm phân vị 0.4 và 0.6 của giá:\n', df['Price'].quantile([0.4, 0.6]))

Giá trung bình các xe:  7645.0
Giá trị trung vị các xe:  6625.0
Miền giá trị của mảng là:  18500.0
Miền phân vị của xe:  0.5    6625.0
Name: Price, dtype: float64
Điểm phân vị 0.4 và 0.6 của giá:
 0.4    5750.0
0.6    7000.0
Name: Price, dtype: float64


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [19]:
# Your code here
df = pd.read_csv('data\\tips.csv')
df

Unnamed: 0  total_bill   tip     sex smoker   day    time  size
0             0       16.99  1.01  Female     No   Sun  Dinner     2
1             1       10.34  1.66    Male     No   Sun  Dinner     3
2             2       21.01  3.50    Male     No   Sun  Dinner     3
3             3       23.68  3.31    Male     No   Sun  Dinner     2
4             4       24.59  3.61  Female     No   Sun  Dinner     4
..          ...         ...   ...     ...    ...   ...     ...   ...
239         239       29.03  5.92    Male     No   Sat  Dinner     3
240         240       27.18  2.00  Female    Yes   Sat  Dinner     2
241         241       22.67  2.00    Male    Yes   Sat  Dinner     2
242         242       17.82  1.75    Male     No   Sat  Dinner     2
243         243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 8 columns]

2. Xoá cột `Unnamed: 0`


In [20]:
# Your code here
df = df.drop(columns=['Unnamed: 0'])
df

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [21]:
# Your code here

print('Số dòng của dữ liệu trên là: ', df.shape[0])
print('Số lượng thuộc tính của dữ liệu trên là: ', df.shape[1])

Số dòng của dữ liệu trên là:  244
Số lượng thuộc tính của dữ liệu trên là:  7


4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [22]:
# Your code here
print('Thuộc tính định tính là: sex, smoker, day, time, size')
print('Thuộc tính định lượng là: total_bill, tip')

Thuộc tính định tính là: sex, smoker, day, time, size
Thuộc tính định lượng là: total_bill, tip


5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [23]:
# Your code here
maleQuantity = df.loc[(df['sex'] == 'Male')].shape[0]
print('Số lượng nam là: ',maleQuantity)
femaleQuantity = df.loc[(df['sex'] == 'Female')].shape[0]
print('Số lượng nữ là: ',femaleQuantity)
print('Tỷ lệ nam: ', ((maleQuantity*100)/244))
print('Tỷ lệ nữ: ', ((femaleQuantity*100)/244))


Số lượng nam là:  157
Số lượng nữ là:  87
Tỷ lệ nam:  64.34426229508196
Tỷ lệ nữ:  35.65573770491803


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [86]:
# Your code here
smokerQuantiy = df.loc[(df['smoker'] == 'Yes')].shape[0]
print('Số lượng người hút thuốc là: ', smokerQuantiy)
print('Tỷ lệ người hút thuốc là: ', ((smokerQuantiy*100)/244))


Số lượng người hút thuốc là:  93
Tỷ lệ người hút thuốc là:  38.114754098360656


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [136]:
# Your code here
crowdedDay = df['day'].value_counts().head(1)
print('Ngày trong tuần khách hàng hay đến dùng bữa nhất là: ', crowdedDay)
print('Tỷ lệ là: ', (crowdedDay.max()*100)/244)

Ngày trong tuần khách hàng hay đến dùng bữa nhất là:  Sat    87
Name: day, dtype: int64
Tỷ lệ là:  35.65573770491803


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [137]:
# Your code here
mostTime = df['time'].value_counts().head(1)
print('Nhà hàng phục vụ bữa ăn nhiều nhất là: ', mostTime)

Nhà hàng phục vụ bữa ăn nhiều nhất là:  Dinner    176
Name: time, dtype: int64


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [145]:
# Your code here
maxCustomerGrp = df['size'].max()
print('Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là:',maxCustomerGrp, 'người')

Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là: 6 người


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [176]:
# Your code here
import numpy as np
arr = []

#Tối thứ năm
customerThurDinner = df.loc[(df['day'] == 'Thur') & (df['time'] == 'Dinner')].shape[0]
arr.append(customerThurDinner)
#Trưa thứ năm
customerThurLunch = df.loc[(df['day'] == 'Thur') & (df['time'] == 'Lunch')].shape[0]
arr.append(customerThurLunch)

#Tối thứ sáu
customerFriDinner = df.loc[(df['day'] == 'Fri') & (df['time'] == 'Dinner')].shape[0]
arr.append(customerFriDinner)
#Trưa thứ sáu
customerFriLunch = df.loc[(df['day'] == 'Fri') & (df['time'] == 'Lunch')].shape[0]
arr.append(customerFriLunch)

#Tối thứ bảy
customerSatDinner = df.loc[(df['day'] == 'Sat') & (df['time'] == 'Dinner')].shape[0]
arr.append(customerSatDinner)
#Trưa thứ bảy
customerSatLunch = df.loc[(df['day'] == 'Sat') & (df['time'] == 'Lunch')].shape[0]
arr.append(customerSatLunch)

#Tối thứ CN
customerSunDinner = df.loc[(df['day'] == 'Sun') & (df['time'] == 'Dinner')].shape[0]
arr.append(customerSunDinner)
#Trưa thứ CN
customerSunLunch = df.loc[(df['day'] == 'Sun') & (df['time'] == 'Lunch')].shape[0]
arr.append(customerSunLunch)
print(arr)

print('Trung bình trong một bữa ăn nhà hàng phục vụ',np.mean(arr),'khách')


[1, 61, 12, 7, 87, 0, 76, 0]
Trung bình trong một bữa ăn nhà hàng phục vụ 30.5 khách


---